QtAcademyProject.h

```diff
@@ -27,20 +36,46 @@ class QtAcademyProject : public QObject
   Q_OBJECT
 
   Q_PROPERTY(Esri::ArcGISRuntime::MapQuickView* mapView READ mapView WRITE setMapView NOTIFY mapViewChanged)
+  Q_PROPERTY(int downloadProgress READ downloadProgress NOTIFY downloadProgressChanged)
+  Q_PROPERTY(bool isTracking READ isTracking WRITE setIsTracking NOTIFY isTrackingChanged)
 
 public:
   explicit QtAcademyProject(QObject* parent = nullptr);
   ~QtAcademyProject() override;
 
+  Q_INVOKABLE void createOfflineAreaFromExtent();
+  Q_INVOKABLE void toggleOffline(bool offline);
+
 signals:
   void mapViewChanged();
+  void downloadProgressChanged();
+  void isTrackingChanged();
 
 private:
   Esri::ArcGISRuntime::MapQuickView* mapView() const;
   void setMapView(Esri::ArcGISRuntime::MapQuickView* mapView);
+  void loadOfflineBasemaps();
+  void exportVectorTiles(Esri::ArcGISRuntime::ArcGISVectorTiledLayer* vectorTileLayer);
+  int downloadProgress();
+  bool isTracking() const;
+  void setIsTracking(bool value);
```

Comment:
I suggest using `bool isTracking` for readability. `value` works, but it's generic when reading the code.